### PR TITLE
Update PDVD sim/reco for MC validation samples production

### PR DESCRIPTION
**Pull request by YoannKermaidic**
The PR contains two parts:

1. Bug fixes of G4 stages when passing from v09_81_00d02 to v09_82_00d00
2. FHICL files that will be used to generate MC validation samples of cosmics, radiologicals and 1 GeV/c charged particles.

### CODE REVIEW

**Comment by YoannKermaidic** [comment]
The G4 stage2 hack avoid running into the following error that only appears with v09_82_00d00 (no change in the service & geometry occured):

```
%MSG-s ArtException:  PostEndJob 10-Jan-2024 14:44:40 CET ModuleEndJob
---- EventProcessorFailure BEGIN
  EventProcessor: an exception occurred during current event processing
  ---- ScheduleExecutionFailure BEGIN
    Path: ProcessingStopped.
    ---- OtherArt BEGIN
      ---- GeometryCore BEGIN
        Cryostat #4294967295 does not exist
        The above exception was thrown while processing module IonAndScint/IonAndScint run: 1 subRun: 0 event: 1
      ---- GeometryCore END
    ---- OtherArt END
    Exception going through path simulate
  ---- ScheduleExecutionFailure END
---- EventProcessorFailure END
---- FatalRootError BEGIN
  Fatal Root Error: TTree::SetEntries
  Tree branches have different numbers of entries, eg EventAuxiliary has 0 entries while sim::ParticleAncestryMap_largeant__G4Stage1. has 1 entries.
  ROOT severity: 2000
---- FatalRootError END
%MSG
```

Seems like the `Cryostat` object is not found, hence dropping `LArG4DetectorServicevolCryostat` IonAndScint instance make it work.

**Comment by YoannKermaidic** [comment]
The G4 stage1 bug fix is identical to the one made by Tom J. for protoDUNE (https://github.com/DUNE/dunesw/commit/60d5b97361df1491d383e4323d9ae3831622b8f0)

**Comment by wenqiang-gu** [comment]
In my test, G4 stage1 also failed with dunesw v09_82_00d00. Here is the error message:
```---- EventProcessorFailure BEGIN
  EventProcessor: an exception occurred during current event processing
  ---- ScheduleExecutionFailure BEGIN
    Path: ProcessingStopped.
    ---- ServiceNotFound BEGIN
      Unable to create ServiceHandle.
      Perhaps the FHiCL configuration does not specify the necessary service?
      The class of the service is noted below...
      ---- ServiceNotFound BEGIN
        ServicesManager unable to find the service of type 'geo::Geometry'.
      ---- ServiceNotFound END
      The above exception was thrown while processing module larg4Main/largeant run: 1 subRun: 0 event: 1
    ---- ServiceNotFound END
    Exception going through path simulate
  ---- ScheduleExecutionFailure END
---- EventProcessorFailure END
---- FatalRootError BEGIN
  Fatal Root Error: TTree::SetEntries
  Tree branches have different numbers of entries, eg EventAuxiliary has 0 entries while simb::MCTruths_generator__SinglesGen. has 1 entries.
  ROOT severity: 2000
---- FatalRootError END
```

**Comment by wenqiang-gu** [comment]
I figured out my issue now. I was using the obsolete fhicl: `protodunevd_g4_stgae1.fcl` instead of `protodunevd_refactored_g4_stage1.fcl`.

**Comment by wenqiang-gu** [comment]
Hi @YoannKermaidic, interestingly, I don't see your error for `protodunevd_refactored_g4_stage2.fcl`. I suggest that you can comment out this line: ```
protodunevd_ionandscint_active.Instances:    "LArG4DetectorServicevolTPCActive"
```

File: fcl/protodunevd/g4/protodunevd_refactored_g4_stage2.fcl
Comment (wenqiang-gu) [inline fcl/protodunevd/g4/protodunevd_refactored_g4_stage2.fcl:1]: In fact, I don't see an error as reported.

**Comment by absolution1** [comment]
Hi there.  
We've seen the same error message appearing in recent FD2-VD simulations.  @tomjunk has been debugging the issue

File: fcl/protodunevd/g4/protodunevd_refactored_g4_stage2.fcl
Comment (wenqiang-gu) [reply fcl/protodunevd/g4/protodunevd_refactored_g4_stage2.fcl:1]: Okay. I can reproduce the error with the cosmic generator now: `gen_protodunevd_cosmics.fcl`

**Comment by wenqiang-gu** [comment]
> Hi there. We've seen the same error message appearing in recent FD2-VD simulations. @tomjunk has been debugging the issue

Hi @absolution1 , do you mean the error message from G4 stage 1 or stage 2?

**Review by wenqiang-gu — APPROVED** [review]

**Pull request merged**